src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirOperationDataStore.cs

```diff
@@ -168,30 +168,39 @@ public async Task<ExportJobOutcome> UpdateExportJobAsync(ExportJobRecord jobReco
 
         public async Task<IReadOnlyCollection<ExportJobOutcome>> AcquireExportJobsAsync(ushort maximumNumberOfConcurrentJobsAllowed, TimeSpan jobHeartbeatTimeoutThreshold, CancellationToken cancellationToken)
```

Comment:
>AcquireExportJobsAsync [](http://example.com/codeflow?start=65&length=22)

Generic comment.
I think you approach would solve problem, but leave same problem open for future cases.

In cosmosDB we have approach where we have one object   
`Container` to interact with Cosmos and we have `CosmosContainerProvider` which not return `Container` unless it's initialized. 

I would look into option to do something similar approach.